packages/wallet/src/TxBuilder/buildTx.ts

```diff
@@ -0,0 +1,262 @@
+import { Cardano } from '@cardano-sdk/core';
+import { Observable, firstValueFrom } from 'rxjs';
+
+import { FinalizeTxProps, InitializeTxProps, InitializeTxResult, ObservableWallet } from '../types';
+import {
+  IncompatibleWalletError,
+  MaybeValidTx,
+  OutputBuilder,
+  OutputValidationMinimumCoinError,
+  OutputValidationTokenBundleSizeError,
+  PartialTxOut,
+  SignedTx,
+  TxBodyValidationError,
+  TxBuilder,
+  TxOutValidationError
+} from './types';
+import { ObservableWalletTxOutputBuilder, toOutputValidationError } from './OutputBuilder';
+import {
+  OutputValidator,
+  RewardAccount,
+  StakeKeyStatus,
+  WalletUtilContext,
+  createWalletUtil,
+  deepEquals
+} from '../services';
+import { SignTransactionOptions, TransactionSigner } from '../KeyManagement';
+
+/**
+ * Minimal sub-type of ObservableWallet that is used by TxBuilder.
+ */
+export type ObservableWalletTxBuilderDependencies = Pick<ObservableWallet, 'submitTx'> & {
+  delegation: { rewardAccounts$: Observable<Pick<RewardAccount, 'address' | 'keyStatus'>[]> };
+} & {
+  initializeTx: (
+    props: Pick<InitializeTxProps, 'auxiliaryData' | 'outputs' | 'signingOptions' | 'extraSigners'>
+  ) => ReturnType<ObservableWallet['initializeTx']>;
+} & {
+  finalizeTx: (
+    props: Pick<FinalizeTxProps, 'auxiliaryData' | 'tx' | 'signingOptions' | 'extraSigners'>
+  ) => ReturnType<ObservableWallet['finalizeTx']>;
+} & WalletUtilContext;
+
+interface CreateSignedTxProps {
+  wallet: ObservableWalletTxBuilderDependencies;
+  initializeTxResult: InitializeTxResult;
+  extraSigners?: TransactionSigner[];
+  signingOptions?: SignTransactionOptions;
+  auxiliaryData?: Cardano.AuxiliaryData;
+}
+
+/**
+ * Transactions built with {@link ObservableWalletTxBuilder.build} method, use this method to sign the transaction.
+ *
+ * @returns Promise with `SignedTx` object that has a {@link SignedTx.submit} method that submits the transaction.
+ */
+const createSignedTx = async ({
+  wallet,
+  initializeTxResult,
+  extraSigners,
+  signingOptions,
+  auxiliaryData
+}: CreateSignedTxProps): Promise<SignedTx> => {
```

Comment:
Props seem to be all the same, so I think CreateSignedTxProps type is redundant
```suggestion
}: FinalizeTxProps): Promise<SignedTx> => {
```

Reply:
[9b5e7cb](https://github.com/input-output-hk/cardano-js-sdk/pull/396/commits/9b5e7cbbb5fcb0d2de9dea91ff28c8ce4ae02511)